lib/roo/excelx.rb

```diff
@@ -97,7 +97,7 @@ def sheets
     def sheet_for(sheet)
       sheet ||= default_sheet
       validate_sheet!(sheet)
-      @sheets_by_name[sheet]
+      @sheets_by_name[sheet] || @sheets[sheet - 1]
```

Comment:
I understand the reason behind using (sheet - 1) is to keep things consistent with validate_sheet!, which is a good thing.

But I think it is confusing to have two methods (sheet, sheet_for) with different starting index. i.e. sheet => 0, sheet_for => 1

What about checking the type of sheet parameter before validate_sheet! and manually adjust sheet value (sheet + 1) in case of a integer. This will keep things consistent to both validate_sheet! and sheet.

In long-term we can somehow warn in depreciation warning that we are changing starting index for sheet and sheet_for from 0 to 1, or maybe we won't need this.